src/main/java/net/sf/jabref/XMPUtilMain.java

```diff
@@ -0,0 +1,174 @@
+package net.sf.jabref;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+
+import javax.xml.transform.TransformerException;
+
+import net.sf.jabref.importer.ParserResult;
+import net.sf.jabref.importer.fileformat.BibtexParser;
+import net.sf.jabref.logic.bibtex.BibEntryWriter;
+import net.sf.jabref.logic.bibtex.LatexFieldFormatter;
+import net.sf.jabref.logic.bibtex.LatexFieldFormatterPreferences;
+import net.sf.jabref.logic.xmp.XMPPreferences;
+import net.sf.jabref.logic.xmp.XMPUtil;
+import net.sf.jabref.model.database.BibDatabaseMode;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.preferences.JabRefPreferences;
+
+import org.apache.jempbox.impl.XMLUtil;
+import org.apache.jempbox.xmp.XMPMetadata;
+
+public class XMPUtilMain {
```

Comment:
Standalone --> cli

2016-07-29 18:13 GMT+02:00 Jörg Lenhard <notifications@github.com>:

> In src/main/java/net/sf/jabref/XMPUtilMain.java
> <https://github.com/JabRef/jabref/pull/1642#discussion_r72818522>:
>
> > +
> > +import net.sf.jabref.importer.ParserResult;
> > +import net.sf.jabref.importer.fileformat.BibtexParser;
> > +import net.sf.jabref.logic.bibtex.BibEntryWriter;
> > +import net.sf.jabref.logic.bibtex.LatexFieldFormatter;
> > +import net.sf.jabref.logic.bibtex.LatexFieldFormatterPreferences;
> > +import net.sf.jabref.logic.xmp.XMPPreferences;
> > +import net.sf.jabref.logic.xmp.XMPUtil;
> > +import net.sf.jabref.model.database.BibDatabaseMode;
> > +import net.sf.jabref.model.entry.BibEntry;
> > +import net.sf.jabref.preferences.JabRefPreferences;
> > +
> > +import org.apache.jempbox.impl.XMLUtil;
> > +import org.apache.jempbox.xmp.XMPMetadata;
> > +
> > +public class XMPUtilMain {
>
> Since most of the code is inside a main method, you are right. I would
> still put it in the cli package, since it otherwise clutters up the root
> package, though. What is the opinion of others? @JabRef/developers
> <https://github.com/orgs/JabRef/teams/developers>
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/JabRef/jabref/pull/1642/files/beef5143cd7160b98c2333bbce97433094c1dc56#r72818522>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATi5P2JH1a2-VzrJQt0zcpZ4paCg2PUks5qaiaZgaJpZM4JYKGO>
> .
>